From jhubbard@nvidia.com  Mon Jul 15 14:10:26 2024
From: John Hubbard <jhubbard@nvidia.com>
Date: Fri, 12 Jul 2024 16:51:50 -0700
Subject: [PATCH] selftests/net: fix gro.c compilation failure due to non-existent opt_ipproto_off
To: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Cc: "Shuah Khan" <shuah@kernel.org>, "David S . Miller" <davem@davemloft.net>, "Eric Dumazet" <edumazet@google.com>, "Jakub Kicinski" <kuba@kernel.org>, "Paolo Abeni" <pabeni@redhat.com>, "Steffen Klassert" <steffen.klassert@secunet.com>, "Herbert Xu" <herbert@gondor.apana.org.au>, "Andreas Färber" <afaerber@suse.de>, "Manivannan Sadhasivam" <manivannan.sadhasivam@linaro.org>, "Matthieu Baerts" <matttbe@kernel.org>, "Mat Martineau" <martineau@kernel.org>, "Geliang Tang" <geliang@kernel.org>, "Pravin B Shelar" <pshelar@ovn.org>, "Willem de Bruijn" <willemdebruijn.kernel@gmail.com>, "Alexander Mikhalitsyn" <alexander@mihalicyn.com>, zhujun2 <zhujun2@cmss.chinamobile.com>, "Petr Machata" <petrm@nvidia.com>, "Ido Schimmel" <idosch@nvidia.com>, "Hangbin Liu" <liuhangbin@gmail.com>, "Nikolay Aleksandrov" <razor@blackwall.org>, "Benjamin Poirier" <bpoirier@nvidia.com>, "Sebastian Andrzej Siewior" <bigeasy@linutronix.de>, "Dmitry Safonov" <0x7f454c46@gmail.com>, netdev@vger.kernel.org, linux-arm-kernel@lists.infradead.org, linux-actions@lists.infradead.org, mptcp@lists.linux.dev, dev@openvswitch.org, linux-kselftest@vger.kernel.org, LKML <linux-kernel@vger.kernel.org>, llvm@lists.linux.dev, "John Hubbard" <jhubbard@nvidia.com>, stable@vger.kernel.org, "Ignat Korchagin" <ignat@cloudflare.com>
Message-ID: <20240712235150.99175-1-jhubbard@nvidia.com>

From: John Hubbard <jhubbard@nvidia.com>

Linux 6.6 does not have an opt_ipproto_off variable in gro.c at all (it
was added in later kernel versions), so attempting to initialize one
breaks the build.

Fixes: c80d53c484e8 ("selftests/net: fix uninitialized variables")
Cc: <stable@vger.kernel.org> # 6.6
Reported-by: Ignat Korchagin <ignat@cloudflare.com>
Closes: https://lore.kernel.org/all/8B1717DB-8C4A-47EE-B28C-170B630C4639@cloudflare.com/#t
Signed-off-by: John Hubbard <jhubbard@nvidia.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 tools/testing/selftests/net/gro.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/tools/testing/selftests/net/gro.c
+++ b/tools/testing/selftests/net/gro.c
@@ -113,9 +113,6 @@ static void setup_sock_filter(int fd)
 		next_off = offsetof(struct ipv6hdr, nexthdr);
 	ipproto_off = ETH_HLEN + next_off;
 
-	/* Overridden later if exthdrs are used: */
-	opt_ipproto_off = ipproto_off;
-
 	if (strcmp(testname, "ip") == 0) {
 		if (proto == PF_INET)
 			optlen = sizeof(struct ip_timestamp);
